contracts/access/manager/AccessManager.sol

```diff
@@ -878,4 +890,11 @@ contract AccessManager is Context, Multicall, IAccessManager {
     function _isExpired(uint48 timepoint) private view returns (bool) {
         return timepoint + expiration() <= Time.timestamp();
     }
+
+    /**
+     * @dev Extracts the selector from calldata. Panics if data is not at least 4 bytes
+     */
+    function _checkSelector(bytes calldata data) private pure returns (bytes4) {
```

Comment:
Why not `_getSelector`?

Reply:
We've been using consistently a convention for reverting functions prefixing them with `_check*`.

I think that sort of makes visually clearer that it reverts (panics really).

Reply:
I guess my question is because those functions usually don't return anything.

It would make sense for me if `_getSelector` reverts when it's not possible to get a selector. Curious about @Amxx's thoughts.

Reply:
My thought was the opposite because in this case I think is more important to make the panic more evident. Even if we don't guarantee an error interface yet, we should try to be as clear as possible.

I'd also like to hear @Amxx's opinion

Reply:
I don't like this function, because it will panic (revert) in some cases where IMO it should just says no. This will cause some "getter" functions to revert, which imo is bad.

IMO the only place where its unclear is in cancel. 

Please see [this diff](https://github.com/OpenZeppelin/openzeppelin-contracts/compare/master...Amxx:openzeppelin-contracts:fv/AccessManager#diff-d33289f3eb8b3043d73cb1803745e64ae1338991933016a3230321df1045d38b) (subset of the FV PR)

Reply:
> This will cause some "getter" functions to revert, which imo is bad.

Can you give a concrete example? I couldn't find one getter function reverting because of that.

> IMO the only place where its unclear is in cancel.

Yeah `cancel` is definitely a place, but I'd consider inconsistent the current `schedule` function. This is because even when it's supposed to revert with an `AccessManagerUnauthorizedCall` it panics.

Also, using only `bytes4` pads the result, which can be misleading (eg. `AccessManagerUnauthorizedCall` reports a different selector to what it was actually used).

Reply:
> I guess my question is because those functions usually don't return anything.

@frangio, it's a similar case with `_useAndCheckNonce`. We can call it `_getAndCheckSelector`.

Reply:
We decided to keep `_checkSelector` after a meeting.